org.eclipse.jdt.ui/ui/org/eclipse/refactoring/MakeStaticAction.java

```diff
@@ -0,0 +1,151 @@
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+import org.eclipse.jface.text.ITextSelection;
+
+import org.eclipse.ui.IWorkbenchSite;
+import org.eclipse.ui.PlatformUI;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.ITypeRoot;
+import org.eclipse.jdt.core.JavaModelException;
+
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringAvailabilityTester;
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringCoreMessages;
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringExecutionStarter;
+import org.eclipse.jdt.internal.corext.util.JavaModelUtil;
+
+import org.eclipse.jdt.ui.actions.SelectionDispatchAction;
+
+import org.eclipse.jdt.internal.ui.IJavaHelpContextIds;
+import org.eclipse.jdt.internal.ui.JavaPlugin;
+import org.eclipse.jdt.internal.ui.actions.ActionUtil;
+import org.eclipse.jdt.internal.ui.actions.SelectionConverter;
+import org.eclipse.jdt.internal.ui.javaeditor.JavaEditor;
+import org.eclipse.jdt.internal.ui.javaeditor.JavaTextSelection;
+import org.eclipse.jdt.internal.ui.refactoring.RefactoringMessages;
+import org.eclipse.jdt.internal.ui.util.ExceptionHandler;
+
+/**
+ * Extracts an expression into a constant field and replaces all occurrences of the expression with
+ * the new constant.
+ *
+ * <p>
+ * This class may be instantiated; it is not intended to be subclassed.
+ * </p>
+ *
+ * @since 3.29
+ *
+ * @noextend This class is not intended to be subclassed by clients.
+ */
+public class MakeStaticAction extends SelectionDispatchAction {
+
+	private JavaEditor fEditor;
+
+	/**
+	 * Note: This constructor is for internal use only. Clients should not call this constructor.
+	 *
+	 * @param editor the java editor
+	 *
+	 * @noreference This constructor is not intended to be referenced by clients.
+	 */
+	public MakeStaticAction(JavaEditor editor) {
+		super(editor.getEditorSite());
+		setText(RefactoringCoreMessages.MakeStaticRefactoring_name);
+		fEditor= editor;
+		setEnabled(SelectionConverter.getInputAsCompilationUnit(fEditor) != null);
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(this, IJavaHelpContextIds.MAKE_STATIC_ACTION);
+	}
+
+	public MakeStaticAction(IWorkbenchSite site) {
+		super(site);
+		setText(RefactoringMessages.MakeStaticAction_title);
+		setToolTipText(RefactoringMessages.MakeStaticAction_tooltip);
+		setDescription(RefactoringMessages.MakeStaticAction_description);
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(this, IJavaHelpContextIds.MAKE_STATIC_ACTION);
+	}
+
+	// Selected in Editor window
+	@Override
+	public void selectionChanged(ITextSelection selection) {
+		setEnabled(fEditor != null && SelectionConverter.getInputAsCompilationUnit(fEditor) != null);
```

Comment:
Is it valid that the refactoring is initialized with a `null` editor? And in that case, can this method be called at all, as not editor exists?

Reply:
I deleted the part "fEditor != null" because there is no way to call this method when the editor is null as far as I know.